bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,256 @@
+        - type: <class 'generator'>
+    - list(compilation_commands[0].arguments)[1:-1]:
+        - Convert the generator object to list, and extract compiler arguments
+        - 0th element is the compiler name
+        - nth element is the filename
+    """
 
-def generate_parsed_info(cursor, filter, lines, this_filename, depth, parsed_list):
-    if cursor.spelling:
-        holder = {
-            "depth": depth,
-            "line": cursor.location.line,
-            "column": cursor.location.column,
-            "kind": cursor.kind.name,
-            "name": cursor.spelling,
-        }
-        if cursor.type.kind.spelling != "Invalid":
-            holder["element_type"] = cursor.type.kind.spelling
-        if cursor.access_specifier.name != "INVALID":
-            holder["access_specifier"] = cursor.access_specifier.name
-        if cursor.result_type.spelling != "":
-            holder["result_type"] = cursor.result_type.spelling
-        if cursor.brief_comment:
-            holder["brief_comment"] = cursor.brief_comment
-        if cursor.raw_comment:
-            holder["raw_comment"] = cursor.raw_comment
-        holder["members"] = []
+    # Extracting argument list from the command's generator object
+    compilation_commands = list(compilation_commands[0].arguments)[1:-1]
 
-        parsed_list.append(holder)
+    return compilation_commands
 
-    for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            child_list = []
-            generate_parsed_info(
-                child, filter, lines, this_filename, depth + 1, child_list,
-            )
-            if child_list and parsed_list:
-                if len(parsed_list[0]["members"]):
-                    parsed_list[0]["members"].append(child_list[0])
-                else:
-                    parsed_list[0]["members"] = child_list
 
+def parse_file(source, compilation_database_path=None):
+    """
+    Returns the parsed_info for a file
 
-def parse_arguments(args):
-    parser = argparse.ArgumentParser(description="C++ libclang parser")
-    parser.add_argument("files", nargs="+", help="The source files to search")
-    return parser.parse_args(args)
+    Arguments:
+        - source: Source to parse
+        - compilation_database_path: The path to `compile_commands.json`
 
+    Returns:
+        - parsed_info (dict)
+    """
 
-def get_output_path(source, output_dir):
-    x_list = source.split("pcl/", 1)[-1]
-    x_list = x_list.split("/")
-    extra = ["pcl", "include"]
+    # Create a new index to start parsing
+    index = clang.Index.create()
 
-    filename = x_list[-1].split(".")[0]
-    relative_dir = "/".join(x for x in x_list[:-1] if x not in extra)
-    dir = os.path.join(output_dir, relative_dir)
+    # Is this check needed?
+    if compilation_database_path is None:
+        # Default compiler argument
+        compilation_commands = ["-std=c++14"]
```

Comment:
This was for when tests didn't have a mock compile_commands.json iirc, so you could remove it here or leave it and remove it in #6.